leshan-core/src/main/java/org/eclipse/leshan/core/oscore/AeadAlgorithm.java

```diff
@@ -10,17 +10,17 @@
 public class AeadAlgorithm implements Serializable {
     private static final long serialVersionUID = 1L; // TODO not sure we will keep SecurityInfo serializable
 
-    public static final AeadAlgorithm AES_GCM_128 = new AeadAlgorithm("A128GCM", 1); //
-    public static final AeadAlgorithm AES_GCM_192 = new AeadAlgorithm("A192GCM", 2); //
-    public static final AeadAlgorithm AES_GCM_256 = new AeadAlgorithm("A256GCM", 3); //
-    public static final AeadAlgorithm AES_CCM_16_64_128 = new AeadAlgorithm("AES-CCM-16-64-128", 10); //
-    public static final AeadAlgorithm AES_CCM_16_64_256 = new AeadAlgorithm("AES-CCM-16-64-256", 11); //
-    public static final AeadAlgorithm AES_CCM_64_64_128 = new AeadAlgorithm("AES-CCM-64-64-128", 12); //
-    public static final AeadAlgorithm AES_CCM_64_64_256 = new AeadAlgorithm("AES-CCM-64-64-256", 13); //
-    public static final AeadAlgorithm AES_CCM_16_128_128 = new AeadAlgorithm("AES-CCM-16-128-128", 30); //
-    public static final AeadAlgorithm AES_CCM_16_128_256 = new AeadAlgorithm("AES-CCM-16-128-256", 31); //
-    public static final AeadAlgorithm AES_CCM_64_128_128 = new AeadAlgorithm("AES-CCM-64-128-128", 32); //
-    public static final AeadAlgorithm AES_CCM_64_128_256 = new AeadAlgorithm("AES-CCM-64-128-256", 33); //
+    public static final AeadAlgorithm AES_GCM_128 = new AeadAlgorithm("A128GCM", 1, 12); //
+    public static final AeadAlgorithm AES_GCM_192 = new AeadAlgorithm("A192GCM", 2, 12); //
+    public static final AeadAlgorithm AES_GCM_256 = new AeadAlgorithm("A256GCM", 3, 12); //
```

Comment:
About nonce size for GCM, I'm not 100% sure.
The [rfc8152#section-10.1](https://datatracker.ietf.org/doc/html/rfc8152#section-10.1) says : 
>  The GCM mode is parameterized by the size of the authentication tag
   and the size of the nonce.  This document fixes the size of the nonce
   at 96 bits.  The size of the authentication tag is limited to a small
   set of values.  For this document however, the size of the
   authentication tag is fixed at 128 bits.
   
@rikard-sics , that sounds good to you ?

Reply:
The underlying COSE library that we use (but modified) uses a nonce size of 96 bits for GCM. This can be seen here:
https://github.com/cose-wg/COSE-JAVA/blob/e461e5e9f87e18f88ff33d5344c7512b4c3b4bdd/src/main/java/COSE/EncryptCommon.java#L30

So 96 bits for the nonce sounds good to me.